src/main/java/com/jvm_bloggers/core/rss/SyndFeedProducer.java

```diff
@@ -1,160 +1,38 @@
 package com.jvm_bloggers.core.rss;
 
-import com.google.common.collect.ImmutableMap;
-import com.jvm_bloggers.core.data_fetching.http.ProtocolSwitchingAwareConnectionRedirectHandler;
+import com.jvm_bloggers.core.rss.fetchers.Fetcher;
 import com.jvm_bloggers.core.utils.Validators;
 import com.rometools.rome.feed.synd.SyndFeed;
-import com.rometools.rome.io.SyndFeedInput;
-import com.rometools.rome.io.XmlReader;
+
+import javaslang.collection.Stream;
 import javaslang.control.Option;
-import lombok.extern.slf4j.Slf4j;
-import org.apache.commons.io.IOUtils;
-import org.springframework.stereotype.Component;
+import javaslang.control.Try;
 
-import java.io.BufferedInputStream;
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.URL;
-import java.net.URLConnection;
-import java.security.KeyManagementException;
-import java.security.NoSuchAlgorithmException;
-import java.security.cert.X509Certificate;
-import java.util.Map;
-import java.util.UUID;
-import java.util.zip.GZIPInputStream;
-import java.util.zip.ZipException;
-import javax.net.ssl.HttpsURLConnection;
-import javax.net.ssl.SSLContext;
-import javax.net.ssl.TrustManager;
-import javax.net.ssl.X509TrustManager;
+import lombok.extern.slf4j.Slf4j;
 
-import static javaslang.control.Option.of;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
 @Component
 @Slf4j
 public class SyndFeedProducer {
 
-    private static final String FAKE_USER_AGENT = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) "
-        + "AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2227.1 Safari/537.36";
+    private final Stream<Fetcher> fetchers;
 
-    private final ProtocolSwitchingAwareConnectionRedirectHandler redirectHandler =
-        new ProtocolSwitchingAwareConnectionRedirectHandler();
+    @Autowired
+    public SyndFeedProducer(java.util.List<Fetcher> fetchers) {
+        this.fetchers = Stream.ofAll(fetchers);
+    }
 
-    /**
-     * Try to fetch rss from given url.
-     * 1st attempt - standard https connection
-     * 2nd attempt - fallback for some self-signed certificates using https connection with
-     * disabled SSL certificate checking
-     * 3rd attempt - execute wget unix command, save as rss a file in tmp directory
-     * and create SyndFeed from it
-     */
     public Option<SyndFeed> createFor(String rssUrl) {
-
-        return createInADefaultWay(rssUrl)
-            .orElse(() -> createWithoutSslVerification(rssUrl))
-            .orElse(() -> createWithWget(rssUrl));
+        return fetchers
+            .map(fetcher -> fetcher.fetch(rssUrl))
+            .find(Try::isSuccess)
```

Comment:
As long as this is `Stream` and not `List` it would be lazily evaluated. This is also covered by unit tests.